.github/workflows/go.yml

```diff
@@ -0,0 +1,39 @@
+name: Go
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+  - cron: '00 16 * * 1'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
```

Comment:
Should we consider [getting the go version from the go mod instead](https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file)?

Reply:
Good call!